tasks/redhat.task/kickstart.erb

```diff
@@ -11,7 +11,20 @@ rootpw <%= node.root_password %>
 network --hostname <%= node.hostname %>
 firewall --enabled --ssh
 authconfig --enableshadow --passalgo=sha512 --enablefingerprint
-timezone --utc America/Los_Angeles
+timezone --utc <%= node.metadata['timezone'] || 'America/Los_Angeles' %>
+rhel_username="<%= node.metadata['rhn_username'] %>"
```

Comment:
Lines 15 - 27 need to go into a %post script. See the [RHEL docs](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Installation_Guide/sect-kickstart-syntax.html#sect-kickstart-postinstall) for details on the syntax.

In a nutshell, only specific kickstart directives are legal in the main part of a kickstart file; to run a shell script, you need to enclose it in `%post ... %end`